### PR TITLE
Adds the ability to specify aliases for datasources of frames

### PR DESCRIPTION
**Pull request by rushan-zagidullin**
New feature was implemented according to our needs and theese questions
https://www.cuba-platform.ru/discuss/t/psevdonimy-dlya-istochnikov-dannyh-vlozhennyh-frejmov/268
https://www.cuba-platform.ru/discuss/t/peredacha-datasource-vo-frame/666/2

Usage:
```xml
<frame id="myFrame" screen="myFrame" width="100%">
    <dsContext>
        <alias name="animalsDs" datasource="dogsDs"/>
    </dsContext>
</frame>
<frame id="myFrame2" screen="myFrame" width="100%">
    <dsContext>
        <alias name="animalsDs" datasource="catsDs"/>
    </dsContext>
</frame>
```

myFrame.xml:
```xml
<window xmlns="http://schemas.haulmont.com/cuba/window.xsd"
        class="com.company.name.gui.MyFrame"
        messagesPack="com.company.name.gui">
    <layout>
        <table id="animalsTable">
            <columns>
                <column id="name"/>
            </columns>
            <rows datasource="animalsDs"/>
        </table>
    </layout>
</window>
```

### CODE REVIEW

File: modules/gui/src/com/haulmont/cuba/gui/xml/layout/loaders/FrameComponentLoader.java [hunk 1]
```diff
@@ -125,4 +127,20 @@ public void loadComponent() {
             loadDescriptorWatch.stop();
         }
     }
+
+    protected void loadAliases() {
+        Element dsAliases = element.element("dsContext");
+        if (dsAliases != null && frameLoader instanceof FrameLoader) {
+            ComponentLoaderContext frameLoaderInnerContext = ((FrameLoader) frameLoader).getInnerContext();
+            for (Object o : dsAliases.elements("alias")) {
```
Comment (knstvk) [inline modules/gui/src/com/haulmont/cuba/gui/xml/layout/loaders/FrameComponentLoader.java]: Use `com.haulmont.bali.util.Dom4j#elements(org.dom4j.Element, java.lang.String)`

**Comment by knstvk** [comment]
Also, what is the motivation of using two nested elements? 
```
<frame id="myFrame" screen="myFrame" width="100%">
    <dsContext>
        <alias name="animalsDs" datasource="dogsDs"/>
    </dsContext>
</frame>
```
I would prefer the structure suggested in [this topic](https://www.cuba-platform.ru/discuss/t/psevdonimy-dlya-istochnikov-dannyh-vlozhennyh-frejmov/268):
```
<frame id="myFrame" screen="myFrame" width="100%">
    <dsAlias datasource="dogsDs" alias="animalsDs"/>
</frame>
```

**Comment by rushan-zagidullin** [comment]
I wanted to group alias elements, because it is possible that there will be other groups of sub elements for other goals. If you will decide to choose short version, I will restructure scheme

**Comment by knstvk** [comment]
Let's go for the short variant. It will look decent even if other nested elements appear.

**Comment by rushan-zagidullin** [comment]
Done

**Comment by knstvk** [comment]
Thank you!

**Pull request merged**